bootstore/src/schemes/v0/fsm_output.rs

```diff
@@ -0,0 +1,157 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Output related types for V0 protocol state machine methods
+
+use super::messages::{
+    Envelope, Error, Request, RequestType, Response, ResponseType,
+};
+use crate::trust_quorum::{RackSecret, TrustQuorumError};
+use sled_hardware::Baseboard;
+use std::collections::BTreeSet;
+use uuid::Uuid;
+
+/// Output from a a call to [`Fsm::Handle`]
+#[derive(Debug, PartialEq, Eq)]
+pub struct Output {
+    /// Possible state that needs persisting before any messages are sent
```

Comment:
It looks like we never have `(persist | envelopes) & api_output`, which suggests that this could be an `enum` instead, e.g.
```rust
pub enum Output {
    Persist { persist: bool },
    Request { request: Vec<Envelope> },
    PersistAndRequest { persist: bool, request: Vec<Envelope> },
    ApiOutput(Result<ApiOutput, ApiError>),
}
```

This has fewer degrees of freedom, although I don't like needing separate variants for `Persist / Request / PersistAndRequest`.  I'd be curious whether we could structure the code so that you never simultaneously need to both `Persist` and `Request`, to remove the `PersistAndRequest` variant.

(If not, I'd be tempted to remove the separate `Persist` and `Request` variants, and just always use the `PersistAndRequest` variant; this would still be an improvement over the status quo)

Reply:
No longer relevant as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)